replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ReplicatorHistoryImpl.java

```diff
@@ -82,15 +125,49 @@ public ReplicatorHistoryImpl(
   @Override
   public void addReplicationEvent(ReplicationStatus replicationStatus) {
     try {
+      ReplicationStatus previous =
+          getReplicationEvents(replicationStatus.getReplicatorName())
+              .stream()
+              .findFirst()
+              .orElse(null);
+      if (previous != null) {
+        addStats(previous, replicationStatus);
+        provider.update(new UpdateRequestImpl(previous.getId(), getMetacardFromStatus(previous)));
+        return;
+      }
+      replicationStatus.setLastRun(replicationStatus.getStartTime());
+      if (replicationStatus.getStatus().equals(Status.SUCCESS)) {
+        replicationStatus.setLastSuccess(replicationStatus.getStartTime());
+      }
       framework.create(new CreateRequestImpl(getMetacardFromStatus(replicationStatus)));
+
     } catch (IngestException | SourceUnavailableException e) {
-      LOGGER.warn(
-          "Error creating replication history item for {}",
-          replicationStatus.getReplicatorName(),
+      throw new ReplicationPersistenceException(
+          "Error creating replication history item for " + replicationStatus.getReplicatorName(),
           e);
     }
   }
 
+  @Override
+  public void removeReplicationEvent(String replicatorid) {
+    try {
+      provider.delete(new DeleteRequestImpl(replicatorid));
+    } catch (IngestException e) {
+      throw new ReplicationPersistenceException(
+          "Error deleting replication history item " + replicatorid, e);
+    }
+  }
+
+  @Override
+  public void removeReplicationEvents(Set<String> ids) {
+    try {
```

Comment:
if you keep the current implementation, you could pass each id to removeReplicationEvent and reduce this method

Reply:
Ignore 👆 if you change removeReplicationEvent

Reply:
This delete will perform much better than calling the other remove method. I think I am going to just change the other method.